Automattic/wp-calypso
# Emails: Clearly distinguish between domains eligible for free trials and other introductory offers

There are no visual changes. This effort mostly attempts to correct the implementation of `isDomainEligibleForTitanFreeTrial()` and `isDomainEligibleForGoogleWorkspaceFreeTrial()` and to differentiate a discounted introductory offer from one that offers a free trial.

#### Proposed Changes

* Introduce new domain functions: `isDomainEligibleForTitanIntroductoryOffer()`, `isDomainEligibleForGoogleWorkspaceIntroductoryOffer()`
* Introduce new product functions: `hasIntroductoryOffer()`, `hasIntroductoryOfferFreeTrial()`
* Refactor existing functions to correctly identify free trials: `isDomainEligibleForTitanFreeTrial()`, `isDomainEligibleForGoogleWorkspaceFreeTrial()`
* Add product parameter to function references that need to determine the existence of free trials
* Refactor references in tests

#### Testing Instructions



* Add a new domain to the cart. Confirm that email upsell comes up i.e. at `/domains/add/[:domain]/email/[:site]`
* Confirm that you are able to view the email comparison without visible errors
* Confirm that the free trial offer for Titan is still visible
* Confirm that you can successfully purchase a new subscription for either of Professional Email and Google Workspace
* Attempt the above steps for an existing domain i.e. at `/email/[:domain]/manage/[:site]`


#### Pre-merge Checklist






Related to #
